amethyst_assets/src/loader.rs

```diff
@@ -108,9 +108,9 @@ unsafe impl<A, E> Sync for AssetFutureCell<A, E> {}
 /// `ThreadPool`.
 pub struct Loader {
     contexts: FnvHashMap<TypeId, Box<Any>>,
-    directory: Arc<Box<AnyStore>>,
+    directory: Arc<AnyStore>,
```

Comment:
It even doesn't matter if it allocates. What does is implementation of `CoerceUsized`.